### PR TITLE
kyc/social: use twitter oembed-api

### PR DESCRIPTION
**Pull request by ice-dionysos**

### CODE REVIEW

File: kyc/social/internal/twitter.go [hunk 1]
```diff
@@ -17,7 +19,7 @@ import (
 )
 
 func (*twitterVerifierImpl) VerifyText(doc *goquery.Document, expectedText string) (found bool) {
-	doc.Find("title").EachWithBreak(func(_ int, s *goquery.Selection) bool {
+	doc.Find("p").EachWithBreak(func(_ int, s *goquery.Selection) bool {
```
Comment (ice-ares) [inline kyc/social/internal/twitter.go:22]: whats p?

Comment (ice-dionysos) [reply kyc/social/internal/twitter.go:22]: That's what twitter returns. The text is inside `<p>...</p>` tag.

File: kyc/social/internal/twitter.go [hunk 2]
```diff
@@ -26,12 +28,13 @@ func (*twitterVerifierImpl) VerifyText(doc *goquery.Document, expectedText strin
...
-func (t *twitterVerifierImpl) VerifyPostLink(doc *goquery.Document) (foundPost bool) {
-	doc.Find("*").EachWithBreak(func(_ int, s *goquery.Selection) bool {
+func (t *twitterVerifierImpl) VerifyPostLink(ctx context.Context, doc *goquery.Document) (foundPost bool) {
+	doc.Find("a").EachWithBreak(func(_ int, s *goquery.Selection) bool {
 		for _, node := range s.Nodes {
 			for i := range node.Attr {
-				if node.Attr[i].Key == "href" && node.Attr[i].Val == t.Post {
-					foundPost = true
+				if node.Attr[i].Key == "href" && strings.HasPrefix(node.Attr[i].Val, "https://t.co") {
+					data, err := t.Scrape(ctx, node.Attr[i].Val)
```
Comment (ice-ares) [inline kyc/social/internal/twitter.go:36]: Whats the structure of data?

File: kyc/social/internal/contract.go [hunk 3]
```diff
@@ -64,6 +64,10 @@ type (
 		Countries []string
 	}
 
+	twitterOE struct {
+		HTML string `json:"html"`
```
Comment (ice-ares) [inline kyc/social/internal/contract.go:68]: Can u have this as []byte ? So we avoid all that copying?

File: kyc/social/internal/twitter.go (hunk 2 above)
Comment (ice-dionysos) [reply kyc/social/internal/twitter.go:36]: It's a redirect, our scraper API supports only GET requests, so we need to parse the body to get the location:
```html
<head><noscript><META http-equiv="refresh" {URL}"></noscript>
...
```

File: kyc/social/internal/contract.go (hunk 3 above)
Comment (ice-dionysos) [reply kyc/social/internal/contract.go:68]: JSON parser expects a string here, if I change that to `[]byte` it complains: `illegal base64 data at input byte 0`

File: kyc/social/internal/twitter.go (hunk 2 above)
Comment (ice-ares) [reply kyc/social/internal/twitter.go:36]: and what does that redirect to? Another html?

File: kyc/social/internal/contract.go (hunk 3 above)
Comment (ice-ares) [reply kyc/social/internal/contract.go:68]: Ok

File: kyc/social/internal/twitter.go (hunk 2 above)
Comment (ice-dionysos) [reply kyc/social/internal/twitter.go:36]: It redirects to the original page, or re-post in our case.

**Pull request merged**